### DISCUSSION TITLE
Is the 'Transform data' feature supported?

### DISCUSSION THREAD

**Discussion by pddg** [grafana/grafonnet#198] [Q&A]
Grafana provides powerfull way to manipulate data before visualizing it, called 'Transform data'.
https://grafana.com/docs/grafana/latest/panels-visualizations/query-transform-data/transform-data/
I use this feature a lot in my dashboards (especially when using tables).

I found a PR in the grafonnet-lib that would support this.
https://github.com/grafana/grafonnet-lib/pull/265

And `withTransformations` seems to be available in Grafonnet. 
https://grafana.github.io/grafonnet/API/panel/table/index.html#fn-queryoptionswithtransformations

However, I have not found any examples or documentation of its use.
Is this feature what I am looking for? And how would I use it?

**Comment by rglbr** [comment]
The feature is supported.
See https://grafana.github.io/grafonnet/API/panel/table/queryOptions/transformation.html

You can use `withTransformations` along with tables like below.
```jsonnet
local g = import "github.com/grafana/grafonnet/gen/grafonnet-latest/main.libsonnet"

local table = g.panel.table.new("Title")
+ g.panel.table.withTransformations(
   g.panel.table.queryOptions.transformations. ...
)

```

**Reply by pddg** [reply]
Thank you for your reply.
I am still unsure of the identifier of the transformer and the schema of the corresponding options.
The Grafana documentation assumes a GUI and does not seem to mention anything about representation as JSON.
It would be nice to know the documentation on these, code to refer to, etc. ......

**Comment by rglbr** [comment]
You could create a Dashboard in Grafana UI with all the transformations you need and then at the JSON model from the Grafana UI.

The JSON model will have the identifier for the transformer and other details

________________________________
